### ISSUE TITLE
Validation: jsonpatch.applyPatch() accepts "patches" that are Objects but not Arrays

### ISSUE THREAD

**Issue by jankeromnes** [Starcounter-Jack/JSON-Patch#186]
Steps to reproduce:

```js
try {
  jsonpatch.applyPatch({ a: 5 }, /* patch = */ { content: 'Invalid patch' }, /* validateOperation = */ true);
} catch (error) {
  console.error('Something went wrong:', error);
}
```

- Expected behavior: An error is thrown (something like "Invalid JSON Patch").
- Actual observed behavior: No error is thrown (just like if `patch` was an empty Array).

In the [applyPatch implementation](https://github.com/Starcounter-Jack/JSON-Patch/blob/02187fbc74f3718ac7498f3c9b955df79cc1c3e2/src/core.ts#L327-L336), we can see that if `patch` is an object like `{}` instead of an Array of valid JSON Patch operations, it will get "applied" successfully (because `patch.length` will be `undefined`, and the for-loop won't run).

Maybe it should throw if `validateOperation` is `true`?

**Comment by alshakero** [comment]
Hi! Thank you and I agree, we should error when this happens. Will look at it soon.

**Comment by alshakero** [comment]
Released 2.0.4. Thank you so much @jankeromnes.

**Comment by jankeromnes** [comment]
My pleasure! Thanks for fixing it so quickly 😃
